src/lib/scanner.ts

```diff
@@ -98,116 +104,46 @@ export class ScanRunner {
-                for (const warning of executionResult.warnings) {
-                    // Since (for now) DFA only runs on a single method,
-                    // if we couldn't find that method, then that's a critical
-                    // error even though DFA itself just considered it a warning.
-                    if (warning.toLowerCase().startsWith('no methods in file ')) {
-                        throw new Error(warning);
-                    }
-                }
-            }
-
-            const result: string = executionResult.result;
-
-            if (result.startsWith("<!DOCTYPE")) {
-                // If the results are an HTML body, then violations were found. Return that.
-                return result;
-            } else {
-
-                // Otherwise, violations weren't found. Return an empty string.
-                return "";
-            }
-        } else {
-            // Any other status code indicates an error of some kind.
-            throw new Error(executionResult.message);
-        }
-    }
-
-    /**
-     *
-     * @param executionResult The results from a scan.
-     * @returns The Rule Results pulled out of the execution results, or an empty array.
-     * @throws if {@coder executionResult.status} is non-zero
-     */
-    private processPathlessResults(executionResult: ExecutionResult): RuleResult[] {
-        // 0 is the status code indicating a successful analysis.
-        if (executionResult.status === 0) {
-            // If the results were a string, that indicates that no results were found.
-            // TODO: Maybe change the plugin to return an empty array instead?
-            //       If that happens, this needs to change.
-            if (typeof executionResult.result === 'string') {
-                return [];
-            } else {
-                return executionResult.result;
-            }
-        } else {
-            // Any other status code indicates an error of some kind.
-            throw new Error(executionResult.message);
+    private async invokeAnalyzer(args: Inputs, isDfa: boolean): Promise<AnyJson> {
+        const sfVersion = '0.0.0';
+        const logger: Logger = await Logger.child('vscode');
+        const display:Display = new VSCodeDisplay();
+        const inputProcessor: InputProcessor = new InputProcessorImpl(sfVersion, display);
+        const ruleFilterFactory: RuleFilterFactory = new RuleFilterFactoryImpl();
+        const resultsProcessorFactory: ResultsProcessorFactory = new ResultsProcessorFactoryImpl();
+        if (isDfa) {
+            const engineOptionsFactory: EngineOptionsFactory = new RunDfaEngineOptionsFactory(inputProcessor);
+            const runAction:RunDfaAction = new RunDfaAction(logger, display, inputProcessor, ruleFilterFactory, engineOptionsFactory,
+                resultsProcessorFactory);
+            await runAction.validateInputs(args);
+            return runAction.run(args);
+                    } else {
```

Comment:
The whitespace seems off here compared to line 135.

Reply:
Fixed